### PR TITLE
add php 7.4 support (update jms serializer v3 + phpUnit) + fix issue 41

### PR DESCRIPTION
**Pull request by medliii**
Plus fix [41](https://github.com/allure-framework/allure-php-commons/issues/41) via change

`    
public static function registerAnnotationNamespaces()
    {
        AnnotationRegistry::registerUniqueLoader('class_exists');
    }
`

doctrine/annotations 2 won't support AnnotationRegistry::register*
And this behavior will be by default
https://github.com/doctrine/annotations/pull/205

### CODE REVIEW

File: src/Yandex/Allure/Adapter/Annotation/AnnotationProvider.php [hunk 1]
```diff
@@ -71,18 +71,7 @@ private static function getAnnotationReader()
...
     {
-        AnnotationRegistry::registerAutoloadNamespace(
-            'JMS\Serializer\Annotation',
-            [
-                __DIR__ . "/../../../../../../../../vendor/jms/serializer/src",
-                __DIR__ . "/../../../../vendor/jms/serializer/src"
-            ]
-        );
-
-        AnnotationRegistry::registerAutoloadNamespace(
```
Comment (medliii) [inline src/Yandex/Allure/Adapter/Annotation/AnnotationProvider.php]: This functions are deprecated

**Comment by techhemant** [comment]
I hope this gets fixed ASAP. It's an obstacle for our test runs.

File: test/Yandex/Allure/Adapter/Support/StepSupportTest.php [hunk 2]
```diff
@@ -48,7 +50,7 @@ public function testExecuteStepCorrectly()
     /**
      * @expectedException Exception
      */
-    public function testExecuteFailingStep()
+    public function te1stExecuteFailingStep()
```
Comment (techhemant) [inline test/Yandex/Allure/Adapter/Support/StepSupportTest.php]: Change this back to testExecuteFailingStep

Comment (medliii) [reply test/Yandex/Allure/Adapter/Support/StepSupportTest.php]: fixed

**Comment by techhemant** [comment]
@medliii Do you have proof of it working with real phpunit test? Can you share screenshots of the same?

**Comment by remorhaz** [comment]
Bump! Is there any way for me to help with accepting this PR?

**Comment by remorhaz** [comment]
> why delete composer.lock?
I don't know the original author's intention, but in general it's useless or even harmful to keep composer.lock in libraries. When the package is installed as a library, it's lock file is ignored; that means that "stable" combination of dependencies that was tested is never preserved. So I'd rather advice to delete it, too.

**Comment by baev** [comment]
@remorhaz thanks for clarification

@medliii could you please fix merge conflicts?

**Comment by remorhaz** [comment]
This PR seems to be abandoned, so I've resolved conflicts in another PR #65.

**Comment by baev** [comment]
closed via https://github.com/allure-framework/allure-php-commons/pull/65

**Pull request closed without merging**